src/back/hlsl/writer.rs

```diff
@@ -1441,6 +1566,180 @@ impl<'a, W: Write> Writer<'a, W> {
 
         Ok(())
     }
+
+    /// Helper function that write wrapped function for `Expression::ImageQuery`
+    /// https://docs.microsoft.com/en-us/windows/win32/direct3dhlsl/dx-graphics-hlsl-to-getdimensions
+    fn write_wrapped_image_query_functions(
+        &mut self,
+        module: &Module,
+        func_body: &[crate::Statement],
+        func_ctx: &back::FunctionCtx<'_>,
+    ) -> BackendResult {
+        for sta in func_body.iter() {
+            if let crate::Statement::Emit(ref range) = *sta {
+                for handle in range.clone() {
+                    if let crate::Expression::ImageQuery { image, query } =
```

Comment:
same here, let's bail out instead of adding indentation, it's already quite deep